File: frotend/src/services/recruiterService.js
import api from '../api/axiosConfig';

const recruiterService = {
  createRecruiter: async (recruiterDto) => {
    const response = await api.post('/recruiters', recruiterDto);
    return response.data;
  },
  getRecruiterById: async (id) => {
    const response = await api.get(`/recruiters/${id}`);
    return response.data;
  },
  updateRecruiter: async (id, recruiterDto) => {
    const response = await api.put(`/recruiters/${id}`, recruiterDto);
    return response.data;
  },
  deleteRecruiter: async (id) => {
    const response = await api.delete(`/recruiters/${id}`);
    return response.data;
  },
  getAllRecruiters: async () => {
    const response = await api.get('/recruiters');
    return response.data;
  },
};

export default recruiterService;

File: backend/src/main/java/com/placement/portal/service/RecruiterService.java
package com.placement.portal.service;

import com.placement.portal.dto.RecruiterDto;

import java.util.List;

public interface RecruiterService {
    RecruiterDto createRecruiter(RecruiterDto recruiterDto);
    RecruiterDto getRecruiterById(Long id);
    RecruiterDto updateRecruiter(Long id, RecruiterDto recruiterDto);
    void deleteRecruiter(Long id);
    List<RecruiterDto> getAllRecruiters();
}
